### PR TITLE
[d16-10][CI] Allow to enable or disable the dotnet builds. (#11358) 

### PR DESCRIPTION
**Pull request by mandel-macaque**
Create a new parameter that can be used to decide if we build or not the
dotnet parts of the project. If we do not, we make sure that we do not
have any errors in all the other steps.

Cherry-pick + disable on d16-10 by default.

### CODE REVIEW

File: tools/devops/automation/templates/build/build.yml [hunk 1]
```diff
@@ -274,13 +278,21 @@ steps:
...
     fi
 
-    CONFIGURE_FLAGS="$CONFIGURE_FLAGS --enable-dotnet --enable-dotnet-windows --enable-install-source"
+    if [[ "$EnableDotNet" == "True" ]]; then
+      echo "Enabling dotnet builds."
+      CONFIGURE_FLAGS="$CONFIGURE_FLAGS --enable-dotnet --enable-dotnet-windows"
+    fi
+
+    CONFIGURE_FLAGS="$CONFIGURE_FLAGS --enable-install-source"
+    echo "Configuration falgs are '$CONFIGURE_FLAGS'"
```
Comment (tj-devel709) [inline tools/devops/automation/templates/build/build.yml:287]:
```suggestion
    echo "Configuration flags are '$CONFIGURE_FLAGS'"
```

File: tools/devops/automation/templates/build/sign-and-notarized.yml [hunk 2]
```diff
@@ -86,36 +90,37 @@ steps:
...
-
-    $SignFileList = @{
-        "SignFileRecordList" = $SignFileRecord
-    }
 
-    # Write the json to a file
-    ConvertTo-Json -InputObject $SignFileList -Depth 5 | Out-File -FilePath $(Build.ArtifactStagingDirectory)/MsiFiles2Notarize.json -Force
-    dotnet $Env:MBSIGN_APPFOLDER/ddsignfiles.dll /filelist:$(Build.ArtifactStagingDirectory)/MsiFiles2Notarize.json
-  displayName: 'Sign .msi'
+      # Write the json to a file
```
Comment (tj-devel709) [inline tools/devops/automation/templates/build/sign-and-notarized.yml:120]: Looks like there is some spacing issues here on github

Comment (mandel-macaque) [reply tools/devops/automation/templates/build/sign-and-notarized.yml:120]: It is correct, he indentation grew due to the {{ if }}} but github diff is confusing.

**Review by tj-devel709 — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Closing since we landed everything already.

**Pull request closed without merging**